ipaserver/plugins/idviews.py

```diff
@@ -694,6 +694,20 @@ def resolve_anchor_to_object_name(ldap, obj_type, anchor):
 
                 if type_correct:
                     return name
+        else:
+            # Without the DCERPC bindings the sid is not resolvable, return
+            # ipaAnchorUUID
+            _dn = DN(("cn", api.packages[0].idviews.DEFAULT_TRUST_VIEW_NAME),
+                     api.env.container_views + api.env.basedn)
+            try:
+                entry = ldap.find_entry_by_attr(attr="ipaanchoruuid",
+                                                value=anchor,
+                                                object_class="ipaUserOverride",
+                                                attrs_list=["ipaoriginaluid"],
+                                                base_dn=_dn)
+            except errors.EmptyResult:
+                pass
+            return entry["ipaoriginaluid"][0]
```

Comment:
this can be `entry.single_value("ipaoriginaluid")` and probably better to move return to inside try/except block. Otherwise we'll get an exception for the empty result or missing entry variable.